File: Net5CoreWebAPI/Models/Product.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Net5CoreWebAPI.Models
{
    public class Product
    {
        [Key]
        public int ProductId { get; set; }

        [Required]
        public string ProductName { get; set; }

        public string Code { get; set; }

        [Required]
        public decimal Price { get; set; }

        public int Quantity {get; set;}

        public DateTime DateCreated { get; set; }
    }
}

File: client-app/src/components/Product.js
import React, { useContext } from 'react';
import { Button } from 'react-bootstrap';
import ProductContext from '../context/productContext';

function ProductItem() {
    const context = useContext(ProductContext);
    console.log("Product item", context.product.productId);

    return (
        <>
            <tr>
                <td>{context.product.productId}</td>
                <td>{context.product.productName}</td>
                <td>{context.product.code}</td>
                <td>{context.product.price}</td>
                <td>{context.product.quantity}</td>
                <td>
                    <Button className="btn btn-sm btn-primary mb-2"
                        onClick={() => context.update(context.product.productId)}>Update
                    </Button>

                    &nbsp;&nbsp;
                    <Button className="btn btn-sm btn-danger mb-2"
                        onClick={() => context.delete(context.product.productId)}>Delete
                    </Button>
                </td>
            </tr>
        </>
    )
}

export { ProductItem };